src/storage/UrlContainerManager.ts

```diff
@@ -1,20 +1,21 @@
 import { ContainerManager } from './ContainerManager';
 import { ensureTrailingSlash } from '../util/Util';
 import { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import { RuntimeConfig } from '../init/RuntimeConfig';
 
 /**
  * Determines containers based on URL decomposition.
  */
 export class UrlContainerManager implements ContainerManager {
-  private readonly root: string;
+  private readonly runtimeConfig: RuntimeConfig;
 
-  public constructor(root: string) {
-    this.root = this.canonicalUrl(root);
+  public constructor(runtimeConfig: RuntimeConfig) {
+    this.runtimeConfig = runtimeConfig;
   }
 
   public async getContainer(id: ResourceIdentifier): Promise<ResourceIdentifier> {
     const path = this.canonicalUrl(id.path);
-    if (this.root === path) {
+    if (this.canonicalUrl(this.runtimeConfig.base) === path) {
```

Comment:
This is the only part that felt wrong.
We may want to consider moving `canonicalUrl` into `RuntimeConfig`, but I would keep that for a separate PR in any case.

Reply:
Yea should probably be in the `RuntimeConfig` since we want the base to be consistent for all classes anyway, and we will want to transform the input URL once users can set the base in a config.

Reply:
Check, I'll do that in a next PR.